weco_datascience/__init__.py

```diff
@@ -0,0 +1,15 @@
+import os
+
+"""
+Common functionality for data science applications in the Wellcome Collection
+platform
+"""
+
+# We're using a combination of github releases and actions to publish this
+# package to pypi.
+# We have a workflow set up to publish this package when a new github release
+# is created. Github actions creates a GITHUB_REF environment variable for the
+# build, where GITHUB_REF = the tag ref that triggered the workflow (see https://docs.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables)
+# eg. if a release is tagged 0.1.2, we'll have a GITHUB_REF="0.1.2" env var.
+# Flit then uses this __version__ variable to tag the pypi release.
+__version__ = os.environ["GITHUB_REF"]
```

Comment:
Unfortunately this can't work as the module `__version__` also should be resolvable outside of the build process (eg: `import pandas; print(pandas.__version__)`), and obviously that's an environment variable that isn't usually available.

It [looks like](https://github.com/takluyver/flit/issues/257) the `flit` maintainer doesn't want to add anything that might solve this easily. I've looked around and disappointingly (but perhaps unsurprisingly) this isn't a solved problem in Python - versioneer doesn't work any more, [`dunamai`](https://pypi.org/project/dunamai/) seems to be the modern alternative, or the full kitchen sink packaging tools make it a bit simpler than `flit`. 

tbh this is so much more of a pain than I'd have hoped that, given we want to release manually anyway, shall we just go back to hardcoding it? That's what the `flit` maintainer appears to recommend, I've done it plenty of times before and it's not really an annoyance imo...

Reply:
That's a shame. Silly that I hadn't thought it through.

I'm happy to hard code it - going via release tags seems basically equivalent but with extra steps...

Reply:
i guess it would be possible to create the `__init__.py` from a separate script which grabs the environment variable during the build? feels a bit messy tho

Reply:
Yeah, I'm sure it's possible and not necessarily that hard but it feels like a misdirection of effort. As you say, we're already releasing manually and bumping a version number isn't hard.

Reply:
👍